plugin/ochttp/client_test.go

```diff
@@ -119,3 +120,115 @@ func TestClient(t *testing.T) {
 		}
 	}
 }
+
+func BenchmarkTransportNoInstrumentation(b *testing.B) {
+	benchmarkClientServer(b, &ochttp.Transport{NoStats: true, NoTrace: true})
+}
+
+func BenchmarkTransportTraceOnly(b *testing.B) {
+	benchmarkClientServer(b, &ochttp.Transport{NoStats: true})
+}
+
+func BenchmarkTransportStatsOnly(b *testing.B) {
+	benchmarkClientServer(b, &ochttp.Transport{NoTrace: true})
+}
+
+func BenchmarkTransportAllInstrumentation(b *testing.B) {
+	benchmarkClientServer(b, &ochttp.Transport{})
+}
+
+// Copied from net/http/serve_test.go
+func benchmarkClientServer(b *testing.B, transport http.RoundTripper) {
+	b.ReportAllocs()
+	b.StopTimer()
+	ts := httptest.NewServer(http.HandlerFunc(func(rw http.ResponseWriter, r *http.Request) {
+		fmt.Fprintf(rw, "Hello world.\n")
+	}))
+	defer ts.Close()
+	var client http.Client
+	client.Transport = transport
+	b.StartTimer()
+
+	for i := 0; i < b.N; i++ {
+		res, err := client.Get(ts.URL)
+		if err != nil {
+			b.Fatalf("Get: %v", err)
+		}
+		all, err := ioutil.ReadAll(res.Body)
+		res.Body.Close()
+		if err != nil {
+			b.Fatal("ReadAll:", err)
+		}
+		body := string(all)
+		if body != "Hello world.\n" {
+			b.Fatal("Got body:", body)
+		}
+	}
+
+	b.StopTimer()
+}
+
+func BenchmarkTransportParallel64NoInstrumentation(b *testing.B) {
+	benchmarkClientServerParallel(b, 64, &ochttp.Transport{NoTrace: true, NoStats: true})
+}
+
+func BenchmarkTransportParallel64TraceOnly(b *testing.B) {
+	benchmarkClientServerParallel(b, 64, &ochttp.Transport{NoStats: true})
+}
+
+func BenchmarkTransportParallel64StatsOnly(b *testing.B) {
+	benchmarkClientServerParallel(b, 64, &ochttp.Transport{NoTrace: true})
+}
+
+func BenchmarkTransportParallel64AllInstrumentation(b *testing.B) {
+	benchmarkClientServerParallel(b, 64, &ochttp.Transport{})
+}
+
+func benchmarkClientServerParallel(b *testing.B, parallelism int, transport *ochttp.Transport) {
```

Comment:
Want to call parallelism concurrency or c? We have n number of concurrent goroutines that might be only scheduled on one thread.

Reply:
I agree that concurrency is probably a better fit but the testing package and net/http where this was copied from calls it parallelism so I'm inclined to leave it as-is